src/corehost/src/servicing_index.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "trace.h"
+#include "servicing_index.h"
+
+servicing_index_t::servicing_index_t(const arguments_t& args)
+{
+    m_patch_root = args.svc_dir;
+	if (!m_patch_root.empty())
+	{
+		m_index_file.assign(m_patch_root);
+		append_path(m_index_file, _X("dotnet_servicing_index.txt"));
+	}
+	m_parsed = m_index_file.empty() || !pal::file_exists(m_index_file);
+}
+
+bool servicing_index_t::find_redirection(
+        const pal::string_t& package_name,
+        const pal::string_t& package_version,
+        const pal::string_t& package_relative,
+        pal::string_t* redirection)
+{
+    ensure_redirections();
+
+    pal::stringstream_t stream;
+    stream << package_name << _X("|") << package_version << _X("|") << package_relative;
+
+    auto iter = m_redirections.find(stream.str());
+    if (iter != m_redirections.end())
+    {
+        pal::string_t full_path = m_patch_root;
+        append_path(full_path, iter->second.c_str());
+        if (pal::file_exists(full_path))
+        {
+            *redirection = full_path;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void servicing_index_t::ensure_redirections()
+{
+    if (m_parsed)
+    {
+        return;
+    }
+
+    pal::ifstream_t fstream(m_index_file);
+    if (!fstream.good())
+    {
+        return;
```

Comment:
Caller can query the map, but the map will be empty which can be early outed like in the code above.